lib/galaxy/jobs/handler.py

```diff
@@ -596,7 +597,12 @@ def __filter_jobs_with_invalid_input_states(self, jobs):
                 .filter(
                     or_(
                         model.Dataset.deleted == true(),
-                        model.Dataset.state != model.Dataset.states.OK,
+                        not_(
+                            or_(
+                                model.Dataset.state == model.Dataset.states.OK,
+                                model.Dataset.state == model.Dataset.states.DEFERRED,
+                            )
+                        ),
```

Comment:
```suggestion
                        model.Dataset.state.not_in_((model.Dataset.states.OK, model.Dataset.states.DEFERRED)),
```
It might render to the same SQL, but reads a little easier ?

Reply:
If I'm recalling currently, in my experience this can render a much slower SQL. I'm not saying it won't optimize it - but I've seen past versions of SQL Alchemy implement this very directly and the result was slower. I will change it if you want though 😟.